core/server/worker/src/main/java/alluxio/worker/page/PagedBlockStore.java

```diff
@@ -132,9 +134,11 @@ public static PagedBlockStore create(UfsManager ufsManager, BlockMasterClientPoo
    * @param pageMetaStore meta data store for pages and blocks
    * @param pageSize page size
    */
+  @Inject
   PagedBlockStore(CacheManager cacheManager, UfsManager ufsManager, BlockMasterClientPool pool,
-      AtomicReference<Long> workerId, PagedBlockMetaStore pageMetaStore,
-      long pageSize) {
+                  @Named("workerId") AtomicReference<Long> workerId,
+                  PagedBlockMetaStore pageMetaStore,
+                  @Named("pageSize") Long pageSize) {
```

Comment:
What's the "scope" of a named injection? If I have another component, say `PagedBlockReader` which also takes a `pageSize` parameter, but the page sizes may be different at runtime, how can I specify the page size for each component?

Reply:
If the `pageSize` is a variable, we consider that we should wrap it with all other variables into another object that is used by `PagedBlockStore`.

Reply:
Sure. Does this mean that using named injection should be less preferred than using a dedicated type to contain these configuration options?